modules/http-client/src/main/java/org/jpos/http/client/HttpQuery.java

```diff
@@ -374,6 +374,8 @@ private HttpRequestBase getHttpRequest(Context ctx) {
                 return put;
             case "GET":
                 return new HttpGet(url);
+            case "DELETE":
+                return new HttpDelete(url);
```

Comment:
Why not adding `PATCH` since we are at it?

```suggestion
            case "DELETE":
                return new HttpDelete(url);
            case "PATCH":
                return new HttpPatch(url);
```

Reply:
Shouldn’t PATCH need a request entity body? (Like PUTor POST)

Reply:
You're right, maybe that could be refactored since the setting of the payload is the same for three cases (all except two in fact), maybe that could go in an if? Or a helper method or a lambda that sets the payload. It could go something like this:

In the `switch case` create the request, or just call a factory method that receives the method name and the url and returns the instance.
Set the body if it's not `GET`.
Return.

Or with the lambda could be something like:

```java
    Consumer<HttpEntityEnclosingRequestBase> setBody = (HttpEntityEnclosingRequestBase request) -> {
        String payload = ctx.getString(requestName);
        if (payload != null) {
            request.setEntity(new StringEntity(payload, getContentType(ctx)));
        }
    }

    private HttpRequestBase getHttpRequest(Context ctx) {
        String url = getURL(ctx);
        String payload;
        switch (ctx.getString(methodName)) {
            case "POST":
                HttpPost post = new HttpPost(url);
                setEntity.accept(post);
                return post;
            case "PUT":
                HttpPut put = new HttpPut(url);
                setEntity.accept(put);
                return put;
            case "GET":
                return new HttpGet(url);
            case "DELETE":
                return new HttpDelete(url);
            case "PATCH":
                HttpPatch patch = new HttpPatch(url);
                setEntity.accept(patch);
                return patch;
```

I'm not saying the refactoring should be done, and I'm aware I'm off-topic here :) .

Reply:
FYI: I updated the suggestion.

Reply:
Or just let the switch/case fall through with a little refactoring and comment